lib/matplotlib/ticker.py

```diff
@@ -2256,7 +2256,11 @@ def tick_values(self, vmin, vmax):
             # If we're a minor locator *that expects at least two ticks per
             # decade* and the major locator stride is 1 and there's no more
             # than one minor tick, switch to AutoLocator.
-            return AutoLocator().tick_values(vmin, vmax)
+            ticklocs = AutoLocator().tick_values(vmin, vmax)
+            # Don't overstrike the major labels.
+            ticklocs = [t for t in ticklocs if 
```

Comment:
eh, that needs to use boolean indexing, no?

Reply:
Not sure what you mean?  Your version didn't work because ticklocs is an array and `is_close_to_int` only works on a single value.  This *seems* to work...

Reply:
My patch *also* fixed is_close_to_int to be vectorized: it deleted the lines
```
-    if not np.isfinite(x):
-        return False
```
you may have missed that?
(if `x` is not finite, the expression below will also be false).

Reply:
Ah, my fault.